presto-main/src/main/java/com/facebook/presto/operator/scalar/RowIndeterminateOperator.java

```diff
@@ -0,0 +1,176 @@
+import static com.facebook.presto.sql.gen.SqlTypeBytecodeExpression.constantType;
+import static com.facebook.presto.type.UnknownType.UNKNOWN;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class RowIndeterminateOperator
+        extends SqlOperator
+{
+    public static final RowIndeterminateOperator ROW_INDETERMINATE = new RowIndeterminateOperator();
+
+    private RowIndeterminateOperator()
+    {
+        super(INDETERMINATE, ImmutableList.of(withVariadicBound("T", "row")), ImmutableList.of(), BOOLEAN.getTypeSignature(), ImmutableList.of(parseTypeSignature("T")));
+    }
+
+    @Override
+    public ScalarFunctionImplementation specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        checkArgument(arity == 1, "Expected arity to be 1");
+        Type type = boundVariables.getTypeVariable("T");
+        Class<?> indeterminateOperatorClass = generateIndeterminate(type, functionRegistry);
+        MethodHandle methodHandle = methodHandle(indeterminateOperatorClass, "indeterminate", type.getJavaType(), boolean.class);
+        return new ScalarFunctionImplementation(false, ImmutableList.of(true), ImmutableList.of(true), methodHandle, isDeterministic());
+    }
+
+    private static Class<?> generateIndeterminate(Type type, FunctionRegistry functionRegistry)
+    {
+        CallSiteBinder binder = new CallSiteBinder();
+
+        // Embed the MD5 hash code of input and output types into the generated class name instead of the raw type names,
+        // which could prevent the class name from hitting the length limitation and invalid characters.
+        byte[] md5Suffix = Hashing.md5().hashBytes(type.toString().getBytes()).asBytes();
+
+        ClassDefinition definition = new ClassDefinition(
+                a(PUBLIC, FINAL),
+                CompilerUtils.makeClassName(Joiner.on("$").join("RowIndeterminateOperator", BaseEncoding.base16().encode(md5Suffix))),
+                type(Object.class));
+
+        Parameter value = arg("value", type.getJavaType());
+        Parameter isNull = arg("isNull", boolean.class);
+
+        MethodDefinition method = definition.declareMethod(
+                a(PUBLIC, STATIC),
+                "indeterminate",
+                type(boolean.class),
+                value,
+                isNull);
+
+        Scope scope = method.getScope();
+        BytecodeBlock body = method.getBody();
+
+        Variable wasNull = scope.declareVariable(boolean.class, "wasNull");
+        body.append(wasNull.set(constantFalse()));
+
+        CachedInstanceBinder cachedInstanceBinder = new CachedInstanceBinder(definition, binder);
+
+        LabelNode end = new LabelNode("end");
+        List<Type> fieldTypes = type.getTypeParameters();
+        boolean unknownElement = fieldTypes.stream().anyMatch(t -> t.equals(UNKNOWN));
```

Comment:
The name of this variable is little misleading. Maybe call it "hasUnknownFields"